src/MergePlugin.php

```diff
@@ -314,9 +294,13 @@ public function onPostPackageInstall(PackageEvent $event)
             if ($package === self::PACKAGE_NAME) {
                 $this->logger->info('composer-merge-plugin installed');
                 $this->state->setFirstInstall(true);
-                $this->state->setLocked(
-                    $event->getComposer()->getLocker()->isLocked()
-                );
+                if (version_compare('2.0.0', PluginInterface::PLUGIN_API_VERSION, '>')) {
+                    $this->state->setLocked(
+                        $event->getComposer()->getLocker()->isLocked()
+                    );
+                } else {
+                    $this->state->setLocked(false);
```

Comment:
Find alternative to install inherited dependencies without affecting locked dependencies.

> Also it looks like the plugin now runs `composer update` after the first install?
> This is a problem when running `composer install` on a CI.
> We expect the locked dependencies to be installed as is and not everything to be updated.

_Originally posted by @prudloff-insite in https://github.com/wikimedia/composer-merge-plugin/pull/189#issuecomment-720671800_

Reply:
This could be resolved using `setUpdateAllowList()` with `setAllowListTransitiveDependencies()` in Composer 1 and `setUpdateAllowTransitiveDependencies()` in Composer 2.

Reply:
@prudloff-insite Can you provide me a copy of the `composer.json` and `composer.lock` that should not be updated when requiring the composer-merge-plugin?

Reply:
Here is a minimal Drupal project that reproduces our issue: https://gist.github.com/prudloff-insite/d4523328892937091b5b43f1e95919fc

The first time a contributor calls `composer install` with Composer 2, it will run `composer update` and modify the lock file.
IMHO `composer install` should never modify the lock file, only `composer update` or `composer require` should try to pick up new merged dependencies.

Reply:
Thanks. I'll give this a try as soon as I can.

I did some tests over the weekend and I wasn't able to reproduce the issue despite knowing why it would occur.